### ISSUE TITLE
Sometimes crashes with UTF8 error

### ISSUE THREAD

**Issue by kimtore** [mdrokz/rust-llama.cpp#22]
Some prompts fail with:

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Utf8Error { valid_up_to: 0, error_len: None }', /home/kimt/.cargo/registry/src/index.crates.io-6f17d22bba15001f/llama_cpp_rs-0.3.0/src/lib.rs:528:46

I'm using this model: https://huggingface.co/TheBloke/Luna-AI-Llama2-Uncensored-GGUF

Anything I can do to try to fix this problem?

**Comment by markcda** [comment]
Yes, [see this commit in fork](https://github.com/avxor/rust-llama.cpp/commit/70ec2c9487ef610e3dcbcdec3c2ed4a03858d88d).

**Comment by kimtore** [comment]
@markcda thanks for the workaround. Are some words now going to be dropped from the model's outputif they contain illegal UTF-8 characters?

**Comment by markcda** [comment]
> Are some words now going to be dropped from the model's outputif they contain illegal UTF-8 characters?

The above commit only made specific deployment of the bug. Roughly speaking, the response will be reset if it contains at least one incorrect character. But the application will not crash. To solve your specific problem, I advise you to read the [following post](https://users.rust-lang.org/t/error-handling-and-utf-8-validation/42619/2).
